### DISCUSSION TITLE
Expose Solid Query setStore in setQueryData

### DISCUSSION THREAD

**Discussion by 14gasher** [TanStack/query#4643] [Ideas]
Solid allows us to do fine grained updates with stores. Under the hood, solid query is using stores to store information.

Optimistic updates would be incredibly powerful were we able to do fine grained updates with the set store functionality. This would help prevent unnecessary changes from happening and add a whole new layer of reactivity to an already impressive library.

Just my 2 cents haha.

**Comment by PeterDraex** [comment]
I'm also interested in this. Currently, as `setStore` is not exposed, it's impossible to do an optimistic update without re-rendering the while resultset. If I have a list with 1000 items and I modify one property of one item, all 1000 items will get all their components re-rendered from scratch. There is no way to avoid this, right?

**Comment by binajmen** [comment]
2026 and I'm asking myself the same question :) Did anyone found a proper solution ?

To address your remark @PeterDraex, would using `reconcile` solve the issue ?
```ts
const users = useQuery(() => ({
  queryKey: ["users"],
  queryFn: getUsers,
  reconcile: "id", // Tell which property identifies each item
}));
```

**Reply by PeterDraex** [reply]
Yes, I've discovered reconcile after posting that comment. I've used it, but in my experience it's very buggy in SolidQuery. I've made repro for a few bugs, fix were eventually fixed, but it still seems unstable to me

**Reply by binajmen** [reply]
good to know, thank you
